doajtest/unit/test_harvester.py

```diff
@@ -0,0 +1,97 @@
+import itertools
+import json
+import time
+from datetime import date, datetime
+
+from unittest.mock import Mock, patch
+
+from doajtest.fixtures import AccountFixtureFactory, JournalFixtureFactory, ArticleFixtureFactory
+from doajtest.fixtures.harvester import HarvestStateFactory, EPMCFixtureFactory
+from doajtest.helpers import DoajTestCase
+from portality.api.v2.client.client import DOAJv1API
+from portality.core import app
+from portality.tasks.harvester import HarvesterBackgroundTask
+from portality import models
+from portality.tasks.harvester_helpers.epmc import models as h_models
+from portality.lib import httputil
+from portality.api.v2.client import client as doajclient
+from portality.tasks.harvester_helpers import workflow
+from portality.tasks.harvester_helpers.workflow import HarvesterWorkflow
+from portality.models.harvester import HarvestState
+from portality.models.harvester import HarvesterPlugin
+
+
+class TestHarvester(DoajTestCase):
+
+
+    def mock_get_journals_issns(self, string1):
+        return ["1234-5678", "9876-5432"]
+
+
+    def mock_get_url(self, url):
+        if url.startswith("https://www.ebi.ac.uk/europepmc"):
+            return "blabla"
+        else:
+            self.old_get(url)
+
+    def setUp(self):
+        super(TestHarvester, self).setUp()
+
+        self.get_journals_issns_old = HarvesterWorkflow.get_journals_issns
+        HarvesterWorkflow.get_journals_issns = self.mock_get_journals_issns
+
+        self.publisher = models.Account(**AccountFixtureFactory.make_publisher_source())
+        self.journal = models.Journal(**JournalFixtureFactory.make_journal_source())
+        self.publisher.add_journal(self.journal.id)
+        self.publisher.generate_api_key()
+
+        self.publisher.save()
+        self.journal.save()
+        time.sleep(2)
```

Comment:
Now that we figured out why the blocking save wasn't working, I'd go back to using that in the journal save